File: elixir/6ku/Two Sum.ex
# Two Sum
# https://www.codewars.com/kata/52c31f8e6605bcc646000082
defmodule TwoSum do
  @spec two_sum([integer()], integer()) :: {integer(), integer()}
  def two_sum(numbers, target) do
    for {x, y} <- Enum.with_index(numbers), {a, b} <- Enum.reverse(Enum.with_index(numbers)) do
      if x + a == target and y != b, do: {y, b}
    end
    |> Enum.reject(&is_nil/1)
    |> List.first()
  end
end

File: ruby/6ku/Two Sum.rb
# frozen_string_literal: true
# Two Sum
# https://www.codewars.com/kata/52c31f8e6605bcc646000082/train/ruby
#
def two_sum(nums, target)
  nums.each_with_index do |num, i|
    nums.each_with_index do |num2, j|
      return [i, j] if num + num2 == target && i != j
    end
  end
end

p two_sum([1234, 5678, 9012], 14690)